app/logic/Calendar/Invitation/IncomingInvitation.ts

```diff
@@ -35,25 +40,40 @@ export class IncomingInvitation {
     /* else add participant? */
   }
 
-  /** Handle the other two InvitationMessage cases:
-   *  CancelledEvent: the organiser cancelled an incoming meeting
-   *  ParticpantReply: an invitee replied to your outgoing invitation */
+  /** CancelledEvent: the organiser cancelled an incoming meeting */
+  async updateCancelled() {
+    assert(this.invitationMessage && this.invitationMessage != InvitationMessage.Invitation, "Can't update from an invitation");
+    let event = this.calEvent();
+    // Is this action reversible? If so, need to check timestamp.
+    let organizer = event.participants.find(participant => participant.response == InvitationResponse.Organizer);
+    if (organizer) {
+      organizer.response = InvitationResponse.Decline;
+      await event.save();
+    }
+  }
+
+  /** ParticpantReply: an invitee replied to your outgoing invitation */
+  async updateParticipantReply() {
+    assert(this.invitationMessage && this.invitationMessage != InvitationMessage.Invitation, "Can't update from an invitation");
+    let event = this.calEvent();
+    let invitee = this.event.participants.find(participant => participant.response != InvitationResponse.Organizer);
+    let participant = event.participants.find(participant => participant.emailAddress == invitee.emailAddress);
+    let timestamp = this.message.sent; // TODO Use DTSTAMP from ICS
```

Comment:
We don't currently have an event field for that, but once we do, it's quite easy to parse from ICS, as it's always in UTC, but it's in iCal compact format, so you might as well send it through `parseDate`.
```diff
   }
+  if (vevent.entries.dtstamp) {
+    [event.timestamp] = parseDate(vevent.entries.dtstamp[0]);
+  }
   if (vevent.entries.dtstart) {
```

Reply:
Except this line of code never got changed to use `this.event.lastUpdateTime`...

Reply:
I've just done it